index.js

```diff
@@ -586,7 +590,7 @@ function buildArray (context, location) {
   `
 
   functionCode += `
-    if (!Array.isArray(obj)) {
+    if (!Array.isArray(obj) && !(obj != null && (${supportedTypedArrays.map(type => ' obj.constructor.name === \'' + type + '\' ').join('||')}) )) {
```

Comment:
ugh. It would be better to put the array in a "global" scope and make an indexOf by the obj.constructor.name than this imho. 

I think this will slow down array significantly.

Reply:
I am not familiar with perf implications here.

Would Array.includes also be worth testing?